### PR TITLE
tests: add test coverage for NMSettingsConverter

### PR DESCRIPTION
**Pull request by GregoryIvo**
> **Note**: We are using the Conventional Commits convention for our pull request titles. Please take a look at the [PR title format document](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#submitting-the-changes) for the supported [types](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#type) and [scopes](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#scope).

Brief description of the PR. [e.g. Added `null` check on `object` to avoid `NullPointerException`]

**Related Issue:** This PR fixes/closes {issue number}

**Description of the solution adopted:** A more detailed description of the changes made to solve/close one or more issues. If the PR is simple and easy to understand this section can be skipped.

**Screenshots:** If applicable, add screenshots to help explain your solution

**Any side note on the changes made:** Description of any other change that has been made, which is not directly linked to the issue resolution [e.g. Code clean up/Sonar issue resolution]

### CODE REVIEW

File: kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMSettingsConverterTest.java [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+	Map<String, Variant<?>> resultMap;
+
+	Map<String, Map<String, Variant<?>>> internalComparatorAllSettingsMap;
+	Map<String, Map<String, Variant<?>>> resultAllSettingsMap;
+
+	Map<String, Object> internetNetworkPropertiesInstanciationMap;
+
+	NetworkProperties networkProperties;
+
+	@Test(expected = IllegalArgumentException.class)
```
Comment (mattdibi) [inline kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMSettingsConverterTest.java:42]: Same thing as [here](https://github.com/mattdibi/kura/pull/3#discussion_r1095447688)

File: kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMSettingsConverterTest.java [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+
+		} else {
+			internalComparatorMap.put("method", new Variant<>("auto"));
+		}
+
+		if (netStatus.equals("netIPv4StatusEnabledLAN")) {
+			internalComparatorMap.put("ignore-auto-dns", new Variant<>(true));
+			internalComparatorMap.put("ignore-auto-routes", new Variant<>(true));
+		} else if (netStatus.equals("netIPv4StatusEnabledWAN")) {
+			Optional<List<String>> dnsServers = Optional.of(List.of("1.1.1.1"));
```
Comment (mattdibi) [inline kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMSettingsConverterTest.java:221]:
```suggestion
			Optional<List<String>> dnsServers = Optional.of(Arrays.asList("1.1.1.1"));
```

File: kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMSettingsConverterTest.java [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+			if (dnsServers.isPresent()) {
+				internalComparatorMap.put("dns", new Variant<>(List.of(new UInt32(16843009)), "au"));
```
Comment (mattdibi) [inline kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMSettingsConverterTest.java:223]:
```suggestion
				internalComparatorMap.put("dns", new Variant<>(Arrays.asList(new UInt32(16843009)), "au"));
```

**Comment by GregoryIvo** [comment]
Closed in favour of #5

**Pull request closed without merging**